testutil/src/main/java/org/spine3/test/Verify.java

```diff
@@ -184,25 +184,37 @@ public static void fail(String message, Throwable cause) {
     }
 
     /**
-     * Asserts that two floats are not equal concerning a delta. If the expected value is infinity then the delta value
-     * is ignored.
+     * Asserts that two floats are not equal concerning a delta. If the expected value is infinity or NaN then
+     * the delta value is ignored.
      */
     public static void assertNotEquals(String itemName, float notExpected, float actual, float delta) {
         try {
             // handle infinity specially since subtracting to infinite values gives NaN and the
             // the following test fails
-            //noinspection FloatingPointEquality
-            if (Float.isInfinite(notExpected) && notExpected == actual || Math.abs(notExpected - actual) <= delta) {
+            if (areNaNs(notExpected, actual)
+                    || areSameKindOfInfinity(notExpected, actual)
+                    || Math.abs(notExpected - actual) <= delta) {
                 Assert.fail(itemName + SHOULD_NOT_BE_EQUAL + notExpected + '>');
             }
         } catch (AssertionError e) {
             throw mangledException(e);
         }
     }
 
+    private static boolean areNaNs(float first, float second)
+    {
+        return Float.isNaN(first) && Float.isNaN(second);
+    }
+
+    private static boolean areSameKindOfInfinity(float first, float second)
+    {
+        //noinspection FloatingPointEquality
```

Comment:
We don't use comment-based warning suppression, as they are too specific to IDEA. Use an annotation-based approach instead.

Also we (almost) always have a comment next to the `@SuppressWarnings` annotation explaining why it is OK to do that in that case. Have a look at the Spine code, you'll see a lot of cases.